### PR TITLE
MOTOR-771 Use Sphinx 4 on motor.readthedocs.io

### PR DESCRIPTION
**Pull request by ShaneHarvey**
See https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html#using-a-configuration-file

### CODE REVIEW

File: doc/docs-requirements.txt [hunk 1]
```diff
@@ -1,2 +1,5 @@
 tornado
 aiohttp
+Sphinx~=4.1
```
Comment (prashantmital) [inline doc/docs-requirements.txt:3]: What is `~=`?

File: doc/docs-requirements.txt [hunk 2]
```diff
@@ -1,2 +1,5 @@
...
+sphinx_rtd_theme~=0.5
```
Comment (prashantmital) [inline doc/docs-requirements.txt:4]: Are these not implicitly specified by installing `Sphinx`? Why do they need to be explicitly specified now?

Comment (ShaneHarvey) [reply doc/docs-requirements.txt:4]: They are not. These are readthedocs specific dependencies. The docs recommend pinning them: https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html#pinning-dependencies

File: doc/docs-requirements.txt (hunk 1 above)
Comment (ShaneHarvey) [reply doc/docs-requirements.txt:3]:
```
Mopidy-Dirble ~= 1.1        # Compatible release. Same as >= 1.1, == 1.*
```
See https://pip.pypa.io/en/stable/cli/pip_install/#example-requirements-file

**Review by prashantmital — APPROVED** [review]

**Pull request merged**